zerver/migrations/0468_backfill_edit_history_json.py

```diff
@@ -0,0 +1,46 @@
+# Generated by Django 3.2.2 on 2021-05-11 10:00
+import orjson
+from django.db import migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+BATCH_SIZE = 5000
+
+
+def backfill_edit_messages(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    Message = apps.get_model("zerver", "Message")
+    ArchivedMessage = apps.get_model("zerver", "ArchivedMessage")
+    msg_models = [Message, ArchivedMessage]
+
+    for model in msg_models:
+        if not model.objects.filter(edit_history__isnull=False).exists():
+            return
+
+        messages = model.objects.filter(edit_history__isnull=False)
+        id_lower_bound = messages.earliest("id").id
+        id_upper_bound = messages.latest("id").id
+
+        while id_lower_bound <= id_upper_bound:
+            current_id_upper_bound = min(id_lower_bound + BATCH_SIZE, id_upper_bound)
+            with transaction.atomic():
+                messages = model.objects.filter(
+                    edit_history__isnull=False, id__range=(id_lower_bound, current_id_upper_bound)
+                )
```

Comment:
We should probably do a `.select_for_update` here to lock the messages so that another process doesn't update their edit history while this is running? I think it's pretty low probability but also seems correct to me? @alexmv @mateuszmandera FYI.

Reply:
We should be doing this conversion on SQL, not in Python, by using casts, which will side-step the need for locking by being a single update statement, as well as the performance problems of `bulk_update`.  See https://github.com/zulip/zulip/pull/26385/files#diff-98773ea776cd43874e1f8436d135e0ebc3e12d65adeb18c4aaf06c23356b8091R18-R31 for inspiration.

Reply:
I just verified and neither CZO nor Zulip Cloud has any bad data which would prohibit doing this as a cast, so I expect that this has only ever had real JSON in it (unlike the audit logs) so is safe to do this way.

Reply:
Ah cool. That will make this much easier.